src/js/utils/option-utils.jsx

```diff
@@ -84,7 +84,8 @@ export const debounceProductsFetch = (waitTime, minSearchLength, locationId) =>
             productCode: obj.productCode,
             handlingIcons: obj.handlingIcons,
             lotAndExpiryControl: obj.lotAndExpiryControl,
-            label: `${obj.productCode} - ${obj.name}`,
+            label: `${obj.productCode} - ${obj.translatedName ?? obj.name}`,
+            originalLabel: `${obj.productCode} - ${obj.name}`,
```

Comment:
more common way of doing it (as it was already done in some place, I do not exactly remember where) would be to replace `originalLabel` with `hasTranslatedName: !!obj.translatedName` and taking as an example one of your Tooltips from above:

Changing from:
```
<Tooltip
    html={<div className="text-truncate">{option.originalLabel}</div>}
    theme="dark"
    disabled={option.originalLabel === option.label}
    position="top-start"
  >
```
to:
```
<Tooltip
    html={<div className="text-truncate">{option.label}</div>}
    theme="dark"
    disabled={!option.hasTranslatedName}
    position="top-start"
  >
```